UserTools/nhits/nhits.cpp

```diff
@@ -86,9 +83,23 @@ bool NHits::Execute(){
 
   for( std::vector<SubSample>::iterator is=samples.begin(); is!=samples.end(); ++is){
 #ifdef GPU
-  GPU_daq::nhits_execute(is->m_PMTid, is->m_time);
-  m_ss << " qqq qqq Look at " << is - samples.begin();
-  StreamToLog(DEBUG1);
+
+    std::vector<int> trigger_ns;
+    std::vector<int> trigger_ts;
+    std::vector<int> m_time_int;
```

Comment:
```suggestion
    std::vector<int> m_time_int(is->m_time.size());
```
I think this is more efficient, but I might be wrong. My thinking is that without telling it how many entries the vector requires, it might make it in memory space that isn't large enough, and have to move it to bigger spaces (potentially multiple times) as `push_back()` is called. It's probably fine for the `trigger_*` ones, but this one is potentially huge (10,000s)

This construction of this vector should also probably go outside the loop, also for memory stability. Actually maybe even into the `Initalise()` method? And make it a data member (like the `m_` name implies)

Reply:
Mmm, the construction has to be inside the loop because within each loop iteration we need a different vector (of different size). I made it a class member. How does it look now?

Reply:
Hi
Ok I did it, please take a look.
There is the additional problem that the trigger code never knows (a priori) how long is a data chunk it's receiving, so it can't quite estimate truly how many hits it mai encounter in that data chunk.
Anyway, I did it, please take a look.

Reply:
Yeah I made #58 to remind us. I'm happy with this now. Thanks!